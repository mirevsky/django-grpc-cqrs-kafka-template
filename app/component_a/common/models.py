from django.db import models


# Create your models here.
class PersonModel(models.Model):
    query = models.CharField(max_length=60)
    page_number = models.IntegerField()
    result_per_page = models.IntegerField()
