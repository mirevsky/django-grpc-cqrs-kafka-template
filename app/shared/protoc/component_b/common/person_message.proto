syntax = "proto3";

package component_b.common;

message Person {
  string query = 1;
  int32 page_number = 2;
  int32 result_per_page = 3;
}