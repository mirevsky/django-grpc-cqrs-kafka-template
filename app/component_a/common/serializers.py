from django_grpc_framework import proto_serializers

from shared.library.component_a.common import person_message_pb2
from component_a.common.models import PersonModel


class PersonProtoSerializer(proto_serializers.ModelProtoSerializer):
    class Meta:
        model = PersonModel
        proto_class = person_message_pb2.Person
        fields = '__all__'
