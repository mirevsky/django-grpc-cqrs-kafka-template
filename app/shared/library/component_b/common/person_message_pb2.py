# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: component_b/common/person_message.proto
"""Generated protocol buffer code."""
from google.protobuf import descriptor as _descriptor
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()




DESCRIPTOR = _descriptor.FileDescriptor(
  name='component_b/common/person_message.proto',
  package='component_b.common',
  syntax='proto3',
  serialized_options=None,
  create_key=_descriptor._internal_create_key,
  serialized_pb=b'\n\'component_b/common/person_message.proto\x12\x12\x63omponent_a.common\"E\n\x06Person\x12\r\n\x05query\x18\x01 \x01(\t\x12\x13\n\x0bpage_number\x18\x02 \x01(\x05\x12\x17\n\x0fresult_per_page\x18\x03 \x01(\x05\x62\x06proto3'
)




_PERSON = _descriptor.Descriptor(
  name='Person',
  full_name='component_b.common.Person',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  create_key=_descriptor._internal_create_key,
  fields=[
    _descriptor.FieldDescriptor(
      name='query', full_name='component_b.common.Person.query', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=b"".decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='page_number', full_name='component_b.common.Person.page_number', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
    _descriptor.FieldDescriptor(
      name='result_per_page', full_name='component_b.common.Person.result_per_page', index=2,
      number=3, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  serialized_options=None,
  is_extendable=False,
  syntax='proto3',
  extension_ranges=[],
  oneofs=[
  ],
  serialized_start=63,
  serialized_end=132,
)

DESCRIPTOR.message_types_by_name['Person'] = _PERSON
_sym_db.RegisterFileDescriptor(DESCRIPTOR)

Person = _reflection.GeneratedProtocolMessageType('Person', (_message.Message,), {
  'DESCRIPTOR' : _PERSON,
  '__module__' : 'component_b.common.person_message_pb2'
  # @@protoc_insertion_point(class_scope:component_b.common.Person)
  })
_sym_db.RegisterMessage(Person)


# @@protoc_insertion_point(module_scope)
