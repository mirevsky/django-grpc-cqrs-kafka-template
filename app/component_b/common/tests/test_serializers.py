import pytest

from component_b.common.tests.fixtures import *
from component_b.common.models import PersonModel
from component_b.common.serializers import PersonProtoSerializer
from shared.library.component_b.common import person_message_pb2


@pytest.mark.django_db
class TestPersonProtoSerializer:

    def test_person_proto_serializer_query(self, mock_product, snapshot):
        queryset = PersonModel.objects.all()
        serializer = PersonProtoSerializer(queryset, many=True)
        assert serializer.data == snapshot

    def test_person_proto_serializer_validate(self, mock_product, snapshot):
        serializer = PersonProtoSerializer(
            message=person_message_pb2.Person(query='Test serializer validation', page_number=1, result_per_page=1))
        serializer.is_valid()
        assert serializer.data == snapshot
