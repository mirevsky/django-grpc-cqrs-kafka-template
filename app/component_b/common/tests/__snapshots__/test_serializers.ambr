# name: TestPersonProtoSerializer.test_person_proto_serializer_query
  <class 'ReturnList'> [
    <class 'OrderedDict'> {
      'id': 1,
      'page_number': 1,
      'query': 'Test product',
      'result_per_page': 1,
    },
  ]
---
# name: TestPersonProtoSerializer.test_person_proto_serializer_validate
  <class 'ReturnDict'> {
    'page_number': 1,
    'query': 'Test serializer validation',
    'result_per_page': 1,
  }
---
